### ISSUE TITLE
DevOps: Explore Kong as an API gateway

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-registration#1579]
#### Description:

As a developer, I want to setup an API gateway
So I can share the session between multiple frontends, and benefit of other features: rate limits, metrics, etc.

#### Acceptance Criteria:

Given I am a developer
When I deploy the app's helm chart
Then Kong is deployed

Given I am a developer
When I try to access the common API
Then the Kong gateway is hit

#### Development Checklist:

- [ ] Explore Kong
- [ ] Integrate with helm chart
- [ ] Configure it to being just a passthrough for now
- [ ] Document how to implement session sharing
- [ ] Watch resources from APS team: https://bcgov-my.sharepoint.com/:v:/g/personal/james_elson_gov_bc_ca/EbSU-Ga0HCFNvbeJjbYt8rIB_fQrle0VDlxwyMLt5mov5w?referrer=Teams.TEAMS-ELECTRON&referrerScenario=MeetingChicletGetLink.view.view
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Dependencies

- Blocked by
- Blocking

**Comment by pbastia** [comment]
cc @dleard @joshgamache @patriciarussellCAS @hannavovk and other devops folks. Maybe we can do this next sprint?
